### ISSUE TITLE
Termination checker is not aggressive enough. Can it try to inline suspicios calls?

### ISSUE THREAD

**Issue by GoogleCodeExporter** [agda/agda#1166]
```
Agda version 2.3.2.1  from ubuntu 13.10

----------------------------
{-# OPTIONS --termination-depth=1024 #-} -- Doesn't affect anything, I thought 
it would

module terchecker where

open import Data.List
open import Data.Nat

foo : List ℕ → ℕ
foo [] = 1
foo (x ∷ xs) = foo []        -- Problematic calls: foo []

bar : List ℕ → ℕ
bar [] = 2
bar (x ∷ xs) = subproof xs  -- Problematic call: subproof x xs xs
  where subproof = bar      -- Problematic call: bar

----------------------------

Suggestion: can termination checker try to evaluate these calls more 
aggressively taking into account all available arguments of invocation?

foo (x :: z)  clearly  terminates because it calls already defined foo [] which 
doesn't call foo at all.

bar (x :: xs) clearly terminates: `subproof` doesn't do any funky business with 
with/rewrite/whatever, so for all intended purposes checker can forget about 
existence of `subproof` and inline it treat it as `bar (x ∷ xs) = bar xs` 
which terminates.


Or at least can we have something like {-# INLINE_AGGRESSIVELY #-} as milder 
version of {-# NO_TERMINATION_CHECK #-} to force checker to try evaluate 
whatever is being called before complaining?
```

Original issue reported on code.google.com by `guln...@gmail.com` on 2 Jun 2014 at 4:44

**Comment by UlfNorell** [comment]
Automatic inlining takes care of the second case.

**Comment by UlfNorell** [comment]
The first case would be fixed by #1390, so closing this one.
